.github/workflows/ci.yml

```diff
@@ -3,71 +3,292 @@ name: Build/Test/Release
+            extra-args: "--all-features"
+          - os: windows-latest
+            target: i686-pc-windows-msvc
+            release-name: i686-windows
+            cross: false
+            postfix: ".exe"
+            extra-args: "--all-features"
+
+          # Windows GNU TODO: wine has bug now (
+          # - os: ubuntu-latest
+          #   target: x86_64-pc-windows-gnu
+          #   release-name: x86_64-windows-gnu
+          #   cross: true
+          #   postfix: ".exe"
+          #   extra-args: "--all-features"
+
+          # MacOSX
+          - os: macos-12
+            target: x86_64-apple-darwin
+            release-name: x86_64-darwin
+            cross: false
+            extra-args: "--all-features"
+          - os: macos-14
+            target: aarch64-apple-darwin
+            release-name: aarch64-darwin
+            cross: false
+            extra-args: "--all-features"
 
     steps:
-    - uses: actions/checkout@v4
-      with:
-        submodules: recursive
-    - uses: ilammy/setup-nasm@v1
-      if : ${{ matrix.platforms.os == 'windows-2022' }}
-    - uses: dtolnay/rust-toolchain@stable
-      with:
-        toolchain: stable
-        components: rustfmt, clippy
-    - name: Run cargo fmt
-      run: cargo fmt --all -- --check
-    - name: Run cargo check
-      run: cargo check --all --all-features
-    - name: Run cargo clippy
-      run: cargo clippy --all --all-features -- -D warnings
-    - name: Run cargo test on Linux
-      if: ${{ matrix.platforms.os == 'ubuntu-22.04' }}
-      run: cargo test --all --all-features
-    - name: Run cargo test on non LInux
-      env: # this is a bit counter intuitive - some tests relies on Docker to run and are marked not(ci), but macos runner doesn't have Docker installed, so we make it "ci" to bypass those tests 
-        CLASH_RS_CI: 'true'
-      if: ${{ matrix.platforms.os != 'ubuntu-22.04' }}
-      run: cargo test --all --all-features
-    - name: Build artifacts
-      run: sh ./scripts/build.sh "${{ matrix.platforms.target }}" "${{ matrix.static }}"
-    - name: Stash artifacts
-      uses: actions/upload-artifact@v4
-      with:
-        name: clash-${{ matrix.platforms.os }}-${{ matrix.platforms.target }}-${{ matrix.static }}
-        path: target/artifacts/
-
-
-    - name: Create Release
-      if: contains(github.ref, 'tags/v')
-      id: create_release
-      uses: softprops/action-gh-release@v2        
-      with:
-        generate_release_notes: true
-    - name: Upload Release Asset
-      if: contains(github.ref, 'tags/v')
-      id: upload-release-asset 
-      uses: dwenegar/upload-release-assets@v2
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        release_id: ${{ steps.create_release.outputs.id }}
-        assets_path: target/artifacts
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry
```

Comment:
does this work on Windows?

Reply:
Seems it is. https://github.com/Watfaq/clash-rs/actions/runs/8650063287/job/23718607064
Some magic i think.